src/main/java/org/interledger/cryptoconditions/ConditionImpl.java

```diff
@@ -2,17 +2,29 @@
 
 import java.util.EnumSet;
 
+
 import org.interledger.cryptoconditions.Condition;
 import org.interledger.cryptoconditions.ConditionType;
 import org.interledger.cryptoconditions.FeatureSuite;
+import org.interledger.cryptoconditions.encoding.Base64Url;
 
-final class ConditionImpl implements Condition {
+public final class ConditionImpl implements Condition {
```

Comment:
By the time ConditionInputStream creates a new Condition, a class implementing the interface is needed. The main advantage of using a pre-existing implementation class (ConditionImpl or any other), is that we can force parameter checking/data-integrity at build time. (for example that fulfillment maxLength >= 0). This makes the code slightly less generic, but I think the gain is worth it. (An immutable object with data validation at build time adds zero entropy to the system). 

Anonymous classes, from my point of view, can create future problems since they are "difficult" to control. They act like Closures and each anonymous class in each different method will be considered a different instance by the compiler/JVM. What if we decide to add new logic (for example logging or JPA persistance) to the Condition implementation?. In the case of anonymous classes this becomes very tricky and repetitive. Also external frameworks (Spring, Hibernate or Terracota come to my mind) will have a hard time working with anonymous classes.
 
 I think instantiating a concrete implementation is way more simple (even if in a future that the concrete implementation can be injected, either on the constructor or with Spring DI,...). 

Curiously the immutability for the anonymous class is forced by the Java syntax for the enclosing function parameters. So no gains on this point.